lib/parsers/yarn-lock-parse-base.ts

```diff
@@ -0,0 +1,190 @@
+import * as _ from 'lodash';
+import * as pMap from 'p-map';
+
+import {
+  LockfileParser,
+  PkgTree,
+  DepTreeDep,
+  Dep,
+  ManifestFile,
+  getTopLevelDeps,
+  Lockfile,
+  LockfileType,
+  createDepTreeDepFromDep,
+} from './';
+import { eventLoopSpinner } from 'event-loop-spinner';
+import {
+  InvalidUserInputError,
+  OutOfSyncError,
+  TreeSizeLimitError,
+} from '../errors';
+import { config } from '../config';
+
+const EVENT_PROCESSING_CONCURRENCY = 5;
+
+export type YarnLockFileTypes = LockfileType.yarn;
+
+export interface YarnLockDeps {
+  [depName: string]: YarnLockDep;
+}
+
+export interface YarnLockBase<T extends YarnLockFileTypes> {
+  type: string;
+  object: YarnLockDeps;
+  dependencies?: YarnLockDeps;
+  lockfileType: T;
+}
+
+export interface YarnLockDep {
+  version: string;
+  dependencies?: {
+    [depName: string]: string;
+  };
+  optionalDependencies?: {
+    [depName: string]: string;
+  };
+}
+
+export abstract class YarnLockParseBase<T extends YarnLockFileTypes>
+  implements LockfileParser {
+  private treeSize: number;
+  private eventLoopSpinRate = 20;
+
+  constructor(private type: T) {
+    // Number of dependencies including root one.
+    this.treeSize = 1;
+  }
+
+  public abstract parseLockFile(lockFileContents: string): Lockfile;
+
+  public async getDependencyTree(
+    manifestFile: ManifestFile,
+    lockfile: Lockfile,
+    includeDev = false,
+    strict = true,
+  ): Promise<PkgTree> {
+    if (lockfile.type !== this.type) {
+      throw new InvalidUserInputError(
+        'Unsupported lockfile provided. ' +
+          'Please provide `package-lock.json`.',
```

Comment:
yarn.lock? not package-lock.json

Reply:
yarn uses yarn.lock file, not the npm way of package-lock.json

Reply:
This is how it used to be
https://github.com/snyk/nodejs-lockfile-parser/pull/69/files#diff-5259b645479c0f7e10680520bf8e7ecbL90

Reply:
yep think it was wrong then too 😄

Reply:
Actually @gitphill is correct. I now understand the flow better and indeed it should say yarn.lock

Reply:
@gitphill addressed your comment 😃